tasks/system_probe.py

```diff
@@ -1163,28 +1163,59 @@ def generate_minimized_btfs(
         return
 
     ctx.run(f"mkdir -p {output_dir}")
-    for root, dirs, files in os.walk(source_dir):
-        path_from_root = os.path.relpath(root, source_dir)
 
-        for dir in dirs:
-            output_subdir = os.path.join(output_dir, path_from_root, dir)
-            ctx.run(f"mkdir -p {output_subdir}")
+    check_for_ninja(ctx)
+
+    ninja_file_path = os.path.join(ctx.cwd, 'generate-minimized-btfs.ninja')
+    with open(ninja_file_path, 'w') as ninja_file:
+        nw = NinjaWriter(ninja_file, width=180)
+
+        nw.rule(name="decompress_btf", command="tar -xf $in -C $target_directory")
+
+        nw.rule(name="minimize_btf", command="bpftool gen min_core_btf $in $out $input_bpf_programs")
+
+        nw.rule(name="compress_minimized_btf", command="tar -C $tar_working_directory -cJf $out $in && rm $in")
+
+        for root, dirs, files in os.walk(source_dir):
+            path_from_root = os.path.relpath(root, source_dir)
+
+            for d in dirs:
+                output_subdir = os.path.join(output_dir, path_from_root, d)
+                ctx.run(f"mkdir -p {output_subdir}")
+
+            for file in files:
+                if not file.endswith(".tar.xz"):
+                    continue
+
+                btf_filename = file[: -len(".tar.xz")]
+                minimized_btf_path = os.path.join(output_dir, path_from_root, btf_filename)
 
-        for file in files:
-            if not file.endswith(".tar.xz"):
-                continue
+                nw.build(
+                    rule="decompress_btf",
+                    inputs=[os.path.join(root, file)],
+                    outputs=[os.path.join(root, btf_filename)],
+                    variables={
+                        "target_directory": root,
+                    },
+                )
 
-            btf_filename = file[: -len(".tar.xz")]
-            compressed_source_btf_path = os.path.join(root, file)
-            output_btf_path = os.path.join(output_dir, path_from_root, btf_filename)
-            compressed_output_btf_path = output_btf_path + ".tar.xz"
+                nw.build(
+                    rule="minimize_btf",
+                    inputs=[os.path.join(root, btf_filename)],
+                    outputs=[minimized_btf_path],
+                    variables={
```

Comment:
```suggestion
                    implicit=input_bpf_programs,
                    variables={
```

Reply:
Tried this out, it doesn't define `input_bpf_programs` as expected (the value ends up blank): https://gitlab.ddbuild.io/DataDog/datadog-agent/-/jobs/201983689